internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml

```diff
@@ -81,6 +80,37 @@ spec:
       script: |
         #!/usr/bin/env bash
         #
+        isFBCOptIn() {
+          TMPFILE=$(mktemp)
+          PYXIS_URL="https://pyxis.engineering.redhat.com/v1"
+
+          IFS="/" read REGISTRY REPO IMAGE <<< "${1}"
+          IFS=":" read IMAGE TAG <<< "${IMAGE}"
+
+          FETCH_URL="${PYXIS_URL}/repositories/registry/${REGISTRY}/repository/${REPO}/${IMAGE}/tag/${TAG}"
+
+          # strips the last "/tag" in case $TAG is not set
+          [ -z "${TAG}" ] && FETCH_URL=${FETCH_URL%/tag*}
+
+          curl --insecure -v --negotiate -u: "${FETCH_URL}" -o $TMPFILE
+          # prints "false" in case .fbc_opt_in entry is missing
+          jq -e -r '.fbc_opt_in //false' $TMPFILE && rm -f $TMPFILE
+        }
+
+        PULL_SPEC=$(opm render $(params.fbcFragment) | jq -r 'select(.schema == "olm.bundle") | "\(.image)" | split("@")[0]' |uniq)
```

Comment:
you do not handle the case where there might be 2 elements in your query

Reply:
I did not get what you mean @scoheb . Do you mean if the `opm` return a list of images with different hashes? This query does Just like your example but not using `tr` (because it is ugly :D). 

If it is not that, can you elaborate  what I actually did not cover?

Reply:
Or you mean, even after reducing the result I still have more than one image? In this case it sounds like a problem. Would it be possible to have different `opt_in` value for each of the images?

Reply:
It is possible that you may get 2 images after being reduced. This can happen if the image got renamed and/or moved in quay.

If you get 2, then both need to be opt_in.

Reply:
Understood. So it means, for any reason one of them are not opt_in, should the Pipeline fail or continue without publishing the index?

Reply:
Please take a look again @scoheb